src/selectors/entities/channel_categories.ts

```diff
@@ -189,6 +189,98 @@ export function makeFilterAutoclosedDMs(getAutocloseCutoff = getDefaultAutoclose
     );
 }
 
+export function makeFilterAutoclosedDMs(): (state: GlobalState, channels: Channel[], categoryType: string) => Channel[] {
+    return createSelector(
+        (state: GlobalState, channels: Channel[]) => channels,
+        (state: GlobalState, channels: Channel[], categoryType: string) => categoryType,
+        getCurrentChannelId,
+        (state: GlobalState) => state.entities.users.profiles,
+        getCurrentUserId,
+        getMyChannelMemberships,
+        (state: GlobalState) => getInt(state, Preferences.CATEGORY_SIDEBAR_SETTINGS, Preferences.LIMIT_VISIBLE_DMS_GMS),
+        (originalChannels, categoryType, currentChannelId, profiles, currentUserId, myMembers, limitPref) => {
+            if (categoryType !== CategoryTypes.DIRECT_MESSAGES) {
+                // Only autoclose DMs that haven't been assigned to a category
+                return originalChannels;
+            }
+
+            const channels = originalChannels.slice();
+
+            function sortByLastViewedAt(channelA: Channel, channelB: Channel) {
+                const channelALastViewedAt = myMembers[channelA.id]?.last_viewed_at || 0;
+                const channelBLastViewedAt = myMembers[channelB.id]?.last_viewed_at || 0;
+
+                if (channelBLastViewedAt === channelALastViewedAt) {
+                    return 0;
+                }
+
+                if (channelALastViewedAt === 0) {
+                    return -1;
+                }
+
+                if (channelBLastViewedAt === 0) {
+                    return 1;
+                }
+
+                return Math.sign(channelBLastViewedAt - channelALastViewedAt);
+            }
+
+            channels.sort((channelA, channelB) => {
+                if (isUnreadChannel(myMembers, channelA)) {
+                    if (isUnreadChannel(myMembers, channelB)) {
+                        return sortByLastViewedAt(channelA, channelB);
+                    }
+
+                    return -1;
+                }
+
+                if (isUnreadChannel(myMembers, channelB)) {
+                    return 1;
+                }
+
+                return sortByLastViewedAt(channelA, channelB);
```

Comment:
I think you can reorder this a bit to save some typing by checking if they're different first and then handling both of them being either true or false afterwards
```suggestion
                
                if (isUnreadChannel(myMembers, channelA) && !isUnreadChannel(myMembers, channelB)) {
                    return -1;
                } else if (!isUnreadChannel(myMembers, channelA) && isUnreadChannel(myMembers, channelB)) {
                    return 1;
                }

                return sortByLastViewedAt(channelA, channelB);
```